src/components/RefineSerach.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { Box, Text, Button } from '@chakra-ui/react'
+import { useSelector } from 'react-redux';
+import Router, { useRouter } from 'next/router';
+import { RootState } from '../../src/redux/store';
+import Link from 'next/link';
+
+
+
+const RefineSerach = () => {
+    const CategoryValues = useSelector((state: RootState) => state.categories.data);
+    const router = useRouter()
+    const { id } = router.query as { id: string };
+    const Selectcategory = CategoryValues.find((item) => item.id === Number(id));
+    let arrSelectCat = [Selectcategory?.attribute];
+    let itemsOfarrSelect = arrSelectCat[0];
+    console.log(itemsOfarrSelect)
+
+    return (
+        <>
+            {Selectcategory?.attribute
+                ? <Text>Refine Search</Text>
+                : null
+            }
```

Comment:
Usually `attribute` should always be an empty array instead of null. to protect the type. 
and then you check if the array have length you render its items

This also can work

```suggestion
            {Selectcategory?.attribute.length && <Text>Refine Search</Text>
            }
```